src/components/SearchModalBridge/sorting.ts

```diff
@@ -0,0 +1,107 @@
+import { Pair, Token, TokenAmount } from '@swapr/sdk'
+import { CurrencyAmount } from '@swapr/sdk/dist/entities/fractions/currencyAmount'
+import { useCallback, useMemo } from 'react'
+import { useActiveWeb3React } from '../../hooks'
+import { toDXSwapLiquidityToken, useTrackedTokenPairs } from '../../state/user/hooks'
+import { useAllTokenBalances, useTokenBalances } from '../../state/wallet/hooks'
+
+// compare two token amounts with highest one coming first
+function balanceComparator(balanceA?: TokenAmount, balanceB?: TokenAmount) {
+  if (balanceA && balanceB) {
+    return balanceA.greaterThan(balanceB) ? -1 : balanceA.equalTo(balanceB) ? 0 : 1
+  } else if (balanceA && balanceA.greaterThan('0')) {
+    return -1
+  } else if (balanceB && balanceB.greaterThan('0')) {
+    return 1
+  }
+  return 0
+}
+
+function getTokenComparator(balances: {
+  [tokenAddress: string]: TokenAmount | undefined
+}): (tokenA: Token, tokenB: Token) => number {
+  return function sortTokens(tokenA: Token, tokenB: Token): number {
+    // -1 = a is first
+    // 1 = b is first
+
+    // sort by balances
+    const balanceA = balances[tokenA.address]
+    const balanceB = balances[tokenB.address]
+
+    const balanceComp = balanceComparator(balanceA, balanceB)
+    if (balanceComp !== 0) return balanceComp
+
+    if (tokenA.symbol && tokenB.symbol) {
+      // sort by symbol
+      return tokenA.symbol.toLowerCase() < tokenB.symbol.toLowerCase() ? -1 : 1
+    } else {
+      return tokenA.symbol ? -1 : tokenB.symbol ? -1 : 0
```

Comment:
do a quick check on these signs.
I can't tell for sure if this is wrong but -1 is present twice here.

Reply:
Honestly, I'm not an author of this code, I just copied it from `src/components/SearchModal/sorting.ts` and have no idea about motivation behind it. We can check it as a part of some other refactoring but now I'd rather not mess with it.